base/bucket_gocb.go

```diff
@@ -907,41 +912,53 @@ func (bucket CouchbaseBucketGoCB) GetWithXattr(k string, xattrKey string, rv int
 	}()
 	worker := func() (shouldRetry bool, err error, value interface{}) {
 
-		// TODO: Replace with single op that has built-in handling for accessDeleted for xattr retrieval.
 		gocbExpvars.Add("Get", 1)
-		casGoCB, docErr := bucket.Bucket.Get(k, rv)
-		cas = uint64(casGoCB)
+		// First, attempt to get the document and xattr in one shot. We can't set SubdocDocFlagAccessDeleted when attempting
+		// to retrieve the full doc body, so need to retry that scenario below.
+		res, err := bucket.Bucket.LookupIn(k).
+			GetEx(xattrKey, gocb.SubdocFlagXattr).
+			GetEx("", gocb.SubdocFlagNone).
+			Execute()
 
-		// If key not found, continue to attempt to retrieve the xattr only.  Otherwise standard error handling.
-		if docErr != nil && docErr != gocbcore.ErrKeyNotFound {
-			shouldRetry = isRecoverableGoCBError(docErr)
-			return shouldRetry, docErr, cas
-		}
+		if err == nil {
```

Comment:
this is a pretty big if / else if / else .. what about changing to a `switch` and putting the last `else` in a `default` block?